strawberryfields/apps/similarity.py

```diff
@@ -304,7 +304,13 @@ def orbit_cardinality(orbit: list, modes: int) -> int:
     """
     sample = orbit + [0] * (modes - len(orbit))
     counts = list(Counter(sample).values())
-    return int(factorial(modes) / np.prod(factorial(counts)))
+
+    # factorials of numbers larger than 170 do not fit into a int,
+    # hence return float using the qarg `exact=True`
+    if modes > 170:
+        return factorial(modes, exact=True) / np.prod(factorial(counts, exact=True))
+    else:
+        return int(factorial(modes, exact=False) / np.prod(factorial(counts, exact=False)))

```

Comment:
```suggestion

    return int(factorial(modes, exact=False) / np.prod(factorial(counts, exact=False)))
```
The `else` is not needed as the `return` will 'exit' the if condition :)

Reply:
Agree. But isn't the `if/else` more readable?

Reply:
To some extent yes, although I guess it comes down to personal preference! in either case, we are already semi-stuck using pylint's convention here (which I assume follows PEP8)